source/includes/_authentication.md

```diff
@@ -9,41 +9,141 @@ curl "https://api.handshq.com/v1/authentication" \
+
+While requests made to the HandsHQ API require the authentication measures set out out [here](#strategy), we also provide measures to verify that any requests made from HandsHQ can be authenticated.
+
+Such requests are made as a result of subscribing for webhook notifications (e.g. `version_pdf_created`) where we will send a payload containing information about the event. e.g. attributes about a PDF that was just generated for one of your projects.
+
+<aside class='notice'>
+  Please note that while we provide such measures, it is still up to those who wish to receive webhook notifications to implement the checks of authenticity with the strategy described.
+</aside>
+
+Our strategy involves the generation of a signature which is constructed for each request sent by HandsHQ and can be verified by the consumer due to the fact that the only other holder of the shared secret is HandsHQ, the signature also protects against request tamperings, see below for details:
+
+### Signature details
+
+The signature can be found in a custom header called `X-Handshq-Webhook-Signature`.
+
+This value is generated for every webhook event request.
+
+HandsHQ uses a `HMAC hex digest` for the signature value, which has been generated using:
+
+- Hashing Algorithm: `SHA-256`.
+- Secret value: Your HandsHQ `api_token`.
+- Data: The body of the request `application/json; charset=utf-8`.
+
+If you wish to be able to generate a signature value to compare against then you will need to use the same hashing algorithm and encoding with the same secret value and data.
+
+### Example authentication workflow
+
+> Example Ruby-On-Rails controller contents
+
+```ruby
+  
+  before_action :ensure_valid_signature
+
+  def create
+    # consume webhook
+  end
+
+  private
+
+  def ensure_valid_signature
+    # handle invalid signatures, e.g. return a 404 status
+    raise 'oh no - an invalid signature!' unless valid_signature?
+  end
+
+  def valid_signature?
+    # retrieve hex-encoded HMAC digest
+    provided_signature = request.headers['X-Handshq-Webhook-Signature']
+    # generate your own
+    generated_signature = OpenSSL::HMAC.hexdigest('sha256', ENV['MY_HANDSHQ_API_TOKEN'], request.body.read)
+    # compare values
+    Rack::Utils.secure_compare(generated_signature, provided_signature)
+  end
+```
+> Different frameworks will have their own ways of achieving the workflow as described, and will have their own considerations of best practices, for example in this case the use of `#secure_compare` for additional protection against timing attacks.
+
+Once you have an endpoint which is able to receive notifications and have registered an `event subscription` with the URL pointing to this endpoint, you can do the following to verify the signature.
+
+- Obtain the value of the signature from the `X-Handshq-Webhook-Signature` header.
+- Generate a HMAC hex digest with the details as described [here](#signature-details)
```

Comment:
Cool haven't seen this md before, does it just link to the heading of `signature-details`, does it matter that Signature details is an `h3`?

Reply:
Shouldn't do, the id attribute is being automatically added (presumably by slate or some underlying md processor) based off the content of the heading 👍